sklearn/utils/graph.py

```diff
@@ -14,11 +14,19 @@
 from scipy import sparse
 
 from ..metrics.pairwise import pairwise_distances
+from _param_validation import Interval, Integral, validate_params
 
 
 ###############################################################################
 # Path and connected component analysis.
 # Code adapted from networkx
+@validate_params(
+    {
+        "graph": ["array-like", "sparse matrix"],
```

Comment:
Please also update the docstring to replace `ndarray` by `array-like` in this parameter description